### PR TITLE
Assignment9

### PR DESCRIPTION
**Pull request by BHoward85**
I did test on only on users, no navigation test as I didn't see the need, as we went from home -> users page -> add / edit, the path back works through the submit and cancel buttons.

### CODE REVIEW

File: SecretSanta/test/SecretSanta.Web.UITests/UserPageTests.cs [hunk 1]
```diff
@@ -45,5 +46,189 @@ public void CanGetToUsersPage()
...
+
+            Driver.SwitchTo().Alert().Accept();
+
+            //Assert
+            List<string> userNames = page.UserNames;
+            Assert.IsFalse(userNames.Contains(userFullName));
+        }
+
+        [TestMethod]
+        public void CanCancelDeleteUser()
```
Comment (Sturminator) [inline SecretSanta/test/SecretSanta.Web.UITests/UserPageTests.cs]: Nice job testing on cancel delete. I didn't think to do this, I only tested delete.

File: SecretSanta/src/SecretSanta.Web/wwwroot/js/main.3667b1eff82ccded388f.js [hunk 2]
```diff
@@ -0,0 +1,112 @@
+/******/ (function(modules) { // webpackBootstrap
```
Comment (Keboo) [inline SecretSanta/src/SecretSanta.Web/wwwroot/js/main.3667b1eff82ccded388f.js]: Typically generated files inside of wwwroot should be excluded in the .gitignore file

File: SecretSanta/test/SecretSanta.Web.UITests/Pages/EditUsersPage.cs [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+
+namespace SecretSanta.Web.UITests.Pages
+{
+    public class EditUsersPage
+    {
+        public const string Slug = UsersPage.Slug + "/Edit";
+
+        public IWebDriver Driver { get; }
+        
+        public string UserId => Driver.Url.Substring(Driver.Url.LastIndexOf("/") + 1);
```
Comment (Keboo) [inline SecretSanta/test/SecretSanta.Web.UITests/Pages/EditUsersPage.cs]: This appears un-used

File: SecretSanta/test/SecretSanta.Web.UITests/UserPageTests.cs [hunk 4]
```diff
@@ -0,0 +1,248 @@
...
+            //Arrange
+            var rootUri = new Uri(RootUrl);
+            Driver.Navigate().GoToUrl(new Uri(rootUri, UsersPage.Slug));
+            var page = new UsersPage(Driver);
+
+            //Act
+            page.AddUser.Click();
+
+            //Assert that we are at the right place
+            Assert.IsTrue(Driver.Url.EndsWith(AddUsersPage.Slug));
```
Comment (Keboo) [inline SecretSanta/test/SecretSanta.Web.UITests/UserPageTests.cs]: As a general best practice, if you find yourself needing to put an `Assert` in the middle of a unit test, it is usually a sign that you need a second unit test.

**Pull request closed without merging**